### DISCUSSION TITLE
Catch all event handlers

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1159] [Q&A] [answered]
Is the `catch all event handlers` available in socket.io client version 4.6.1 and also can so can custom events be emitted and listened to inside the connect event or the have to be outside the connect event

**Comment by miguelgrinberg** [comment, marked as answer]
You can check the change log to find out when a feature was added. The catch-all was introduced in 5.4.1.

The connect event is a notification when the connection is established. Probably not a good idea to use it for other purposes.

**Reply by Notadumdum** [reply]
Is there any 3rd party library that I can use to catch all events fir version 4.6.1?

**Reply by miguelgrinberg** [reply]
What people did before the catch-all event handler was added is to override the `trigger_event` method in a class-based namespace.

```python
class MyCustomNamespace(socketio.Namespace):
    def trigger_event(self, event_name, sid, *args):
        print(f"{event_name=}, {sid=}")
        if args:
            print(f"data is {args[0]}")


sio.register_namespace(MyCustomNamespace())
```

**Reply by Notadumdum** [reply]
Please can I get the client.html documentation of the version 4.6.1 because I can't seem to navigate from here 
 https://python-socketio.readthedocs.io/en/latest/
